### PR TITLE
Add Icinga notification template imports

### LINKED ISSUE #266: [Bug] The role task for icinga_notification_template has permit 'imports' key, but the module don't
### Description

The role task 'icinga_notification_template' is configured to set the value of 'imports' key if exists, but in the module plugin is not implemented as permitted key and value retrieval to submit in Director

### Reproduction steps

1. Create a icinga_notification_templates variables like documentation to use with the dedicated role task and set the key 'imports' as list on the elements
2. Launch the automation using the task included in the role
3. The task fails with error: `...msg: 'Unsupported parameters for (icinga_notification_template) module: imports....`

### Current Behavior

The module raises error
`...msg: 'Unsupported parameters for (icinga_notification_template) module: imports....`

### Expected Behavior

Correct retrieval of key 'imports' and submit to Director

### Additional information

I just sent the PR for resolution

### PR DESCRIPTION
**Pull request by gianmarco-mameli**
Closes #266, Icinga notification template imports bug 
Let me know if it's ok, I already tried on my Icinga install

### CODE REVIEW

**Comment by gianmarco-mameli** [comment]
Yes, of course. The run give me errors with YQ `Error: unknown shorthand flag: 'a' in -a=append`.
I'm on MacOS Arm64, yq installed via Brew version 4.44.5, maybe it's too recent?

**Comment by schurzi** [comment]
@gianmarco-mameli that changes from #269 are merged now, can you please rebase and fix the linting errors. If you want to try you can also try to update the tests. For this you need to use yq in version 3.4.1, nothing else will work at the moment.

**Comment by gianmarco-mameli** [comment]
I installed the yq version 3.4.1 on a debian container, because via Brew on mac it's only available the latest version.
I launched the script but I think something is not right:
I received an error and 5 files modified, but all empty (only the yaml `---` on top)

Screenshot 2024-11-19 at 17 35 44
Screenshot 2024-11-19 at 17 35 53

**Comment by schurzi** [comment]
nice! The errors you get are actually expected, because you have some minor problems with the code. ;)
see: https://github.com/telekom-mms/ansible-collection-icinga-director/actions/runs/11880503521/job/33129443662#step:3:987

File: plugins/modules/icinga_notification_template.py [hunk 1]
```diff
@@ -108,6 +108,11 @@
     type: bool
     choices: [true, false]
     version_added: '1.25.0'
+  imports:
```
Comment (schurzi) [inline plugins/modules/icinga_notification_template.py:111]: this is messing keys for `type` and `elements`

File: plugins/modules/icinga_notification_template.py [hunk 2]
```diff
@@ -118,6 +123,8 @@
     url_username: "{{ icinga_user }}"
     url_password: "{{ icinga_pass }}"
     object_name: foonotificationtemplate
+        imports:
```
Comment (schurzi) [inline plugins/modules/icinga_notification_template.py:126]: the indentation is wrong.

**Review by schurzi — CHANGES_REQUESTED** [review]

**Comment by gianmarco-mameli** [comment]
My bad sorry! Now seems to be ok, thanks for the support

**Comment by schurzi** [comment]
> My bad sorry! 

No worries! In the end this is what tests are for. ;)

Now we have a problem with our tests. you added the import for `footemplate`in the task that adds `footemplate` this is not deployable :D

The easy solution might be to leave the added `imports` out of the example. This way it will only be shown in docs, which should be enoug.

**Comment by gianmarco-mameli** [comment]
yep, I see the errors. I'm going to fix it now

**Comment by gianmarco-mameli** [comment]
yeah, checks passed

**Review by schurzi — APPROVED** [review]

**Comment by schurzi** [comment]
Thank you! It was a nice experience working with you on that.

**Pull request merged**